### ISSUE TITLE
Reference model for LoRA

### ISSUE THREAD

**Issue by SpaceHunterInf** [huggingface/trl#4309]
Greetings,

Quick question about our LoRA setup: are we initializing a separate reference model during training (e.g., for KL/anchor comparisons), or are we reusing the frozen base model as the reference since the base is frozen under PEFT?

If a separate reference is currently spun up, is there a way to configure the pipeline to reuse the base model as the reference to save memory/compute? Pointers to the relevant config flags or code path would be much appreciated.

Thank you very much.

**Comment by pramodith** [comment]
@SpaceHunterInf I believe this is already in place. In GRPO for e.g. If you're using a PEFT model a separate reference model isn't initialized, as seen https://github.com/huggingface/trl/blob/47b1aa775705a3dc026fdf4d0ab4f7e6c3c18ae0/trl/trainer/grpo_trainer.py#L458-L461 and when self.beta>0 (i.e. we have a KL penalty in our loss) we disable the adapter and get the logits from the frozen model as you suggested, as seen in the code here https://github.com/huggingface/trl/blob/47b1aa775705a3dc026fdf4d0ab4f7e6c3c18ae0/trl/trainer/grpo_trainer.py#L1495.

So you shouldn't need to add any new configs 😄 .
